src/crowdin_client.cpp

```diff
@@ -143,139 +158,331 @@ dispatch::future<void> CrowdinClient::Authenticate()
+            prj->languages.push_back(Language::TryParse(str::to_wstring(langCode)));
+        //TODO: get more until all files gotten (if more than 500)
+        return m_api->get(url + "/files?limit=500");
+    }).then([this, url, prj](json r)
+    {   
+        // Handle project files
+        for(auto& i : r["data"])
+        {
+            const json& d = i["data"];
+            if(d["type"] != "assets")
             {
-                if (i["can_translate"].get<int>() != 0)
-                    prj.languages.push_back(Language::TryParse(str::to_wstring(i["code"])));
+                const json& dir = d["directoryId"],
+                            branch = d["branchId"];
+                prj->files.push_back({
+                    L"/" + str::to_wstring(d["name"]),
+                    d["id"],
+                    dir.is_null() ? NO_ID : dir.get<int>(),
+                    branch.is_null() ? NO_ID : branch.get<int>()
+                });
             }
-            ExtractFilesFromInfo(prj.files, r, L"/");
-            return prj;
-        });
+        }
+        //TODO: get more until all dirs gotten (if more than 500)
+        return m_api->get(url + "/directories?limit=500");
+    }).then([this, url, prj](json r)
+    {
+        // Handle directories
+        struct dir {
+            std::string name;
+            int parentId;
+        };
+        std::map<int, dir> dirs;
+
+        for(const auto& i : r["data"])
+        {
+            const json& d = i["data"],
+                  parent = d["directoryId"];
+            dirs.insert({
+                d["id"],
+                {
+                    d["name"],
+                    parent.is_null() ? NO_ID : parent.get<int>()
+                }
+            });
+        }
+
+        std::stack<std::string> path;
+        for(auto& i : prj->files)
+        {
+            int dirId = i.dirId;
+            while(dirId != NO_ID)
+            {
+                const auto& dir = dirs[dirId];
+                path.push(dir.name);
+                dirId = dir.parentId;
+            }
+            std::string pathStr;
+            while(path.size())
+            {
+                pathStr += '/';
+                pathStr += path.top();
+                path.pop();
+            }
+            if(!pathStr.empty())
+                i.pathName = str::to_wstring(pathStr) + i.pathName;
+        }
+
+        //TODO: get more until all branches gotten (if more than 500)    
+        return m_api->get(url + "/branches?limit=500");
+    }).then([this, url, prj](json r)
+    {
+        // Handle branches
+        std::map<int, std::string> branches;
+
+        for(const auto& i : r["data"])
+        {
+            const json& d = i["data"];
+            branches[d["id"]] = d["name"].get<std::string>();
+        }
+
+        for(auto& i : prj->files)
+            if(i.branchId != NO_ID)
+                i.pathName = "/" + str::to_wstring(branches[i.branchId]) + i.pathName;
```

Comment:
You're using wstring, if you concatenate it with a string literal, it should be a *wide* string literal, i.e. `L"/"`.

Reply:
Addressed b69f4e1c047ddea3d84d135329d655bda9caf9d9